Flaky test StaticArrayEntryTest.testToStringWithNoValue

## Flaky Test

* **Test Name:** `StaticArrayEntryTest.testToStringWithNoValue`
* **Link:** https://github.com/JanusGraph/janusgraph/blob/a852ca89b7a23ccbe95c59800c5c3e8b088afeb9/janusgraph-test/src/test/java/org/janusgraph/diskstorage/util/StaticArrayEntryTest.java#L49
* **Branch:** `master`

### Stack Trace

```
[ERROR] Failures: 

[ERROR]   StaticArrayEntryTest.testToStringWithNoValue:49 expected: <0x35D200728821E4E> but was: <0x035D200728821E4E>

[INFO] 

[ERROR] Tests run: 416, Failures: 1, Errors: 0, Skipped: 2
```

--- a/janusgraph-test/src/test/java/org/janusgraph/diskstorage/util/StaticArrayEntryTest.java
+++ b/janusgraph-test/src/test/java/org/janusgraph/diskstorage/util/StaticArrayEntryTest.java
@@ -39,14 +39,22 @@ public class StaticArrayEntryTest {
     }
 
     @Test
-    public void testToStringWithNoValue() {
-        long l = random.nextLong();
+    public void testRandomLongToStringWithNoValue() {
+        testToStringWithNoValue(random.nextLong());
+    }
+
+    @Test
+    public void testHexLongWithLeadingZeroToStringWithNoValue() {
+        testToStringWithNoValue(Long.MAX_VALUE >> 3);
+    }
+
+    private void testToStringWithNoValue(long l) {
         StaticBuffer b = BufferUtil.getLongBuffer(l);
         long column = b.getLong(0);
         Entry entry = new StaticArrayEntry(b, 8);
 
         final String[] split = entry.toString().split("->");
-        assertEquals(String.format("0x%08X", column), split[0]);
+        assertEquals(String.format("0x%016X", column), split[0]);
         assertEquals("[no value]", split[1]);
     }
 }

